src/lean.mk.in

```diff
@@ -63,10 +64,12 @@ endif
 # make sure the .olean file is newer than the .depend file to prevent infinite make cycles
 	@touch $@
 
+ifndef C_ONLY
 $(TEMP_OUT)/%.c: $(OLEAN_OUT)/%.olean
 	@
+endif
```

Comment:
@leodemoura Just to make you aware of a somewhat significant, but hopefully otherwise invisible, change in the build system in this PR: we now build the stage 0 C code using the same pipeline as in other stages, i.e. the Makefile and leanc, instead of an `add_library` CMake call (see also removed code in `script/update-stage0`). We do this by skipping the .olean -> .c step in the case of stage 0, as seen here, and using the committed .c files instead, which causes the .olean files not to be built at all (which would fail in stage 0, since there is no `lean` executable yet for doing so).
This change ensures the build pipeline is more consistent across stages and simplifies the whole setup.